lib/recursive-methods.rb

```diff
@@ -1,49 +1,134 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the integer given
+# Space complexity: O(n) because the number of times the function is called is 'n' times and placed on the stack
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+  if n < 0
+    raise ArgumentError
+  elsif n == 0
+    return 1
+  else
+    return n * factorial(n-1)
+  end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of string 's'
+# Space complexity: O(n) because the number of times the function reverse is called and placed on the stack
 def reverse(s)
-    raise NotImplementedError, "Method not implemented"
+  if s.length == 0
+    return s
+  else
+    temp_holding = s[0]
+    return reverse(s[1..-1]) + temp_holding
```

Comment:
`s[1..-1]` creates a new string and copies all the elements over to the new string... so this line by itself is O(n).